src/github.com/getlantern/peerscanner/host.go

```diff
@@ -80,33 +87,75 @@ func (h *host) String() string {
  * API for interacting with host
  ******************************************************************************/
 
-// newHost creates a new host for the given name, ip and optional DNS record.
-func newHost(name string, ip string, record *cloudflare.Record) *host {
-	// Cache TLS sessions
-	clientSessionCache := tls.NewLRUClientSessionCache(1000)
-
+// newHost creates a new host for the given name, ip and optional DNS records.
+func newHost(name string, ip string, port string, cflRecord *cloudflare.Record, dspRecord *dnsimple.Record) *host {
 	h := &host{
 		name:         name,
 		ip:           ip,
-		record:       record,
+		port:         port,
+		cflRecord:    cflRecord,
+		dspRecord:    dspRecord,
 		resetCh:      make(chan string, 1000),
 		unregisterCh: make(chan interface{}, 1),
 		statusCh:     make(chan chan *status, 1000),
+		initCfrCh:    make(chan interface{}, 1),
+	}
+
+	if h.isFallback() {
+
+		h.cflGroups = map[string]*cflGroup{
+			RoundRobin: &cflGroup{subdomain: RoundRobin},
+			Fallbacks:  &cflGroup{subdomain: Fallbacks},
+			Peers:      &cflGroup{subdomain: Peers},
+		}
+		h.dspGroups = map[string]*dspGroup{
+			RoundRobin: &dspGroup{subdomain: RoundRobin},
+			Fallbacks:  &dspGroup{subdomain: Fallbacks},
+			Peers:      &dspGroup{subdomain: Peers},
+		}
+		country := fallbackCountry(name)
+		if country != "" {
+			// Add host to country-specific rotation
+			h.cflGroups[country] = &cflGroup{subdomain: country}
+			h.dspGroups[country] = &dspGroup{subdomain: country}
+		}
+	} else {
+		log.Errorf("Somehow adding peer host? %v (%v)", name, ip)
+	}
+
+	return h
+}
+
+func (h *host) resetProxiedClient(port string) {
+
+	var dial func(addr string) (net.Conn, error)
+	if port == "80" {
+		dialer := net.Dialer{Timeout: dialTimeout}
```

Comment:
Yeah I think that sounds right. I don't see any actual bugs in the PR, so we can probably just merge. I do think we should go back and simplify as much as possible after the merge though, as I think things like this could quickly get out of hand and harder to understand if we don't do cleanups as we go.

Sorry this has taken so long -- peerscanner in particular has changed quite a bit since I last worked on it so took me a bit to get back into it.